src/request_executor.cc

```diff
@@ -72,7 +73,10 @@ ResponseAlloc(
     void** buffer_userp, TRITONSERVER_MemoryType* actual_memory_type,
     int64_t* actual_memory_type_id)
 {
-  SharedMemoryManager* shm_pool = reinterpret_cast<SharedMemoryManager*>(userp);
+  std::unique_ptr<SharedMemoryManager> shm_pool(
+      reinterpret_cast<SharedMemoryManager*>(userp));
+
+  ScopedDefer _([&shm_pool] { shm_pool.release(); });
```

Comment:
So the scope defer doesn't release the `shm_pool` anyway, why managing `shm_pool` with unique_ptr?

Reply:
It is because PbMemory::Create expects a unique_ptr<SharedMemoryManger>.

Reply:
Would it make sense to change `PbMemory::Create` signature to expect just a raw pointer? It is not going to take ownership of the pointer anyway. Then in this case we just pass in the `userp` and in other cases where it is a unique pointer, we call `unique_ptr.get()`

Reply:
Makes sense to me. Will create a follow up PR containing this change.